python/tk_multi_loader/model_latestpublish.py

```diff
@@ -309,22 +300,25 @@ def _load_external_data(self):
             # associate the tree view node hash with this node.
             item.setData(tree_view_item_hash, SgLatestPublishModel.ASSOCIATED_TREE_VIEW_ITEM_ROLE)
 
-            # copy across the std fields SG_ASSOCIATED_FIELD_ROLE and SG_DATA_ROLE
-            tree_item_sg_data = tree_view_item.get_sg_data()
-            item.setData(tree_item_sg_data, SgLatestPublishModel.SG_DATA_ROLE)
+            # Extract the Shotgun data and field value from the tree view item.
+            (tree_view_sg_data, field_value) = model_item_data.get_item_data(tree_view_item)
 
-            tree_item_field_data = tree_view_item.data(shotgun_model.ShotgunModel.SG_ASSOCIATED_FIELD_ROLE)
-            item.setData(tree_item_field_data, SgLatestPublishModel.SG_ASSOCIATED_FIELD_ROLE)
+            # Rebuild tree view item field data.
+            # Since we do not care about the "name" value, arbitrarily use "data".
```

Comment:
would it be easier to read this code if there was a big 

```
if tree_view_item comes from a hierarchy model do X else do Y?
```

The `# Since we do not care about the "name" value, arbitrarily use "data".` confuses me. What does it mean? If we don't care about it, why is it there?